### ISSUE TITLE
InfluxDB: Automatically chose retention policy based on time range

### ISSUE THREAD

**Issue by gelicia** [grafana/grafana#73450]
### Discussed in https://github.com/grafana/grafana/discussions/42509

<div type='discussions-op-text'>

<sup>Originally posted by **dennisjac** March  5, 2016</sup>
Hi,
I'm currently looking at how to make Grafana use the aggregated retention policies in InfluxDB.
Issue #3943 apparently adds a way to set the retention policy statically for a query but this only works if you have a hand full of queries. The moment you have more dashboards this is not really a viable option.

What I'd rather like to propose is the ability to add a table in the settings that defines a retention policy based on time ranges. For example you would define the values like this:

1 month => retention_one_value_per_day
1 week => retention_one_value_per_hour
1 day => retention_one_value_per_10s

All queries in all dashboards would then use the retention policy closest to the selected time range as default (which could still be overridden on a per-query basis by explicitly selecting a retention policy there?). If the user doesn't define this table then the default retention policy would always be used.

With this approach most people would be able to set up this table once and then all dashboards would automatically always use the right retention policy for their data which is I think what 99% of the people out there expect. Also for anyone who doesn't specify this value Grafana will behave just as before.
</div>

**Comment by itsmylife** [comment]
@grafana/observability-metrics does not currently have the bandwidth to prioritize development on this issue. If members of the community want to take this issue we will help review any submitted PRs. Please upvote or comment to help us re-prioritize this issue in the future!

**Comment by nicolas17** [comment]
Yeah I would like to ask Grafana why this has not had activity in the last year.
